ibis/expr/operations.py

```diff
@@ -696,6 +708,14 @@ class Mean(Reduction):
     output_type = rules.scalar_output(_mean_output_type)
 
 
+class Quantile(Reduction):
+
+    input_type = [value,
+                  number(name='quantile', allow_boolean=False),
```

Comment:
We don't have a nice way of describing "numeric scalar or array of numeric scalars". I'll put up a PR to do this now.